### PR TITLE
Kubernetes shared MariaDB 

### PR DESCRIPTION
**Pull request by shreddedbacon**
# Checklist
- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [ ] PR title is ready for changelog and subsystem label(s) applied

Explain the **details** for making this change. What existing problem does the pull request solve?

# Closing issues
Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: images/kubectl-build-deploy-dind/build-deploy-docker-compose.sh [hunk 1]
```diff
@@ -88,7 +88,8 @@ do
     if kubectl --insecure-skip-tls-verify -n ${NAMESPACE} get service "$SERVICE_NAME" &> /dev/null; then
       SERVICE_TYPE="mariadb-single"
     # heck if this cluster supports the default one, if not we assume that this cluster is not capable of shared mariadbs and we use a mariadb-single
-    elif svcat --scope cluster get class $MARIADB_SHARED_DEFAULT_CLASS > /dev/null; then
+    # real basic check to see if the mariadbconsumer exists as a kind
+    elif kubectl --insecure-skip-tls-verify -n ${NAMESPACE} get mariadbconsumer &> /dev/null; then
```
Comment (Schnitzel) [inline images/kubectl-build-deploy-dind/build-deploy-docker-compose.sh]: can you use this (adapt to your groups)
```
kubectl api-resources --no-headers --api-group=servicecatalog.k8s.io | grep serviceinstances -q
```
as with ^^ you can make sure that only your api-group is searched through, and not another one

File: images/kubectl-build-deploy-dind/build-deploy-docker-compose.sh [hunk 2]
```diff
@@ -861,7 +864,9 @@ do
   SERVICE_NAME_IMAGE="${MAP_SERVICE_NAME_TO_IMAGENAME[${SERVICE_NAME}]}"
   SERVICE_NAME_IMAGE_HASH="${IMAGE_HASHES[${SERVICE_NAME_IMAGE}]}"
 
-  helm template ${SERVICE_NAME} /kubectl-build-deploy/helmcharts/${SERVICE_TYPE} -f /kubectl-build-deploy/values.yaml --set image="${SERVICE_NAME_IMAGE_HASH}" | outputToYaml
+  if [ "${SERVICE_TYPE}" != "mariadb-shared" ]; then
```
Comment (Schnitzel) [inline images/kubectl-build-deploy-dind/build-deploy-docker-compose.sh]: let's remove this `if`. as the helm chart should just create the exact same CRD again, plus the helmchart might have additional objects in it that are not added yet (like `prebackuppods`)

File: images/kubectl-build-deploy-dind/scripts/exec-kubernetes-create-crd.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+# Check if the MariaDBConsumer exists and create if not
+if kubectl -n ${NAMESPACE} get "${SERVICEBROKER_CLASS}" "${SERVICE_NAME}" &> /dev/null; then
+  echo "${SERVICEBROKER_CLASS} ${SERVICE_NAME} already existing, not attempting to update"
+else
+  # apply the MariaDBConsume from the template
+  cat ${KUBERNETES_SERVICES_TEMPLATE} | \
+    sed "s/{{SERVICE_NAME}}/${SERVICE_NAME}/g" | \
+    sed "s/{{SERVICEBROKER_PLAN}}/${SERVICEBROKER_PLAN}/g" | \
+    kubectl apply -f -
+fi
```
Comment (Schnitzel) [inline images/kubectl-build-deploy-dind/scripts/exec-kubernetes-create-crd.sh]: let's actually use helm templates for this :)

**Review by Schnitzel — CHANGES_REQUESTED** [review]

**Comment by shreddedbacon** [comment]
@Schnitzel this is ready for a second review whenever you're ready

**Pull request merged**